modules/transactions/src/main/java/org/apache/ignite/internal/tx/MismatchingTransactionOutcomeException.java

```diff
@@ -19,15 +19,13 @@
 
 import static org.apache.ignite.lang.ErrorGroups.Transactions.TX_UNEXPECTED_STATE_ERR;
 
-import org.apache.ignite.tx.TransactionException;
-
 /**
  * The exception is thrown when the transaction result differs from the intended one.
  *
  * <p>For example, {@code tx.commit()} is called for a transaction, but the verification logic decided to abort it instead. The transaction
  * will be finished with {@link TxState#ABORTED} and the call to {@code tx.commit()} will throw this exception.
  */
-public class MismatchingTransactionOutcomeException extends TransactionException {
+public class MismatchingTransactionOutcomeException extends TransactionInternalException {
```

Comment:
Semantically it's not internal.

Reply:
Please take a look at https://github.com/apache/ignite-3/pull/4070#discussion_r1678848166